### PR TITLE
chore(deps): update various version

### PR DESCRIPTION
**Pull request by ckipp01**
This pr updates the following:

- Scala 2.13 to the latest 2.13.10
- mill-integrationtest to 0.6.1
- mill.vcs.version to 0.3.0 and remove the stripping of v (done by default now)
- Mill to latest 0.10.8 with a .mill-version file

### CODE REVIEW

**Comment by ckipp01** [comment]
btw total aside, but I noticed that none of these were caught by Steward when they should have been. I created https://github.com/VirtusLab/scala-steward-repos/issues/98 to follow-up.

File: build.sc [hunk 1]
```diff
@@ -1,23 +1,23 @@
...
 
 object `mill-tpolecat` extends Cross[MillTpolecatCross](millVersions: _*)
 class MillTpolecatCross(millVersion: String) extends CrossModuleBase with PublishModule {
-  override def crossScalaVersion = "2.13.7"
+  override def crossScalaVersion = "2.13.10"
   override def artifactSuffix    = s"_mill${millBinaryVersion(millVersion)}" + super.artifactSuffix()
 
   def compileIvyDeps = Agg(ivy"com.lihaoyi::mill-scalalib:$millVersion")
 
-  def publishVersion: T[String] = VcsVersion.vcsState().format(tagModifier = _.dropWhile(_ == 'v'))
```
Comment (ckipp01) [inline build.sc:20]: So stripping the `v` is no longer necessary as this is done by default now. However, it got me wondering. I don't see any git tags associated with this repo. Are you just manually tagging via GitHub or something or how are you releasing?

Comment (DavidGregory084) [reply build.sc:20]: Hi @ckipp01 I've not been so involved with this repo recently but as far as I know we're just using git tags for releasing on tag pushes
image

Comment (joan38) [reply build.sc:20]: Sorry I missed this PR. I did not quiet understand the next steps. Is `publishVersion` defined by default without `v` now?

Comment (ckipp01) [reply build.sc:20]: > Hi @ckipp01 I've not been so involved with this repo recently but as far as I know we're just using git tags for releasing on tag pushes

Yup, this was my bad, I hadn't fetched everything upstream and was only dealing with my fork. Just ignore that part.

> Sorry I missed this PR. I did not quiet understand the next steps. Is publishVersion defined by default without v now?

Well the logic we had here was stripping the `v` which as of 0.2.0 is done by [default](https://github.com/lefou/mill-vcs-version/releases/tag/0.2.0).

**Comment by ckipp01** [comment]
Thanks for the follow-up on this. Just let me know what else you need to get this across the board. More importantly, can you also look at https://github.com/DavidGregory084/mill-tpolecat/pull/18.

**Comment by joan38** [comment]
Thanks @ckipp01 !

**Pull request merged**